### ISSUE TITLE
False positive triggers for `async_without_await`, `unneeded_throws_rethrows` with `override` functions

### ISSUE THREAD

**Issue by akuzminskyi** [realm/SwiftLint#6416]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`async_without_await`, `unneeded_throws_rethrows` "false positive" triggers when function has `override`.

```swift
class A {
    func test() async {
        try? await Task.sleep(for: .seconds(0.5)) 
        print("A")
    }
}

class B: A {
    override func test() async { 
        print("B")
    }
}

Task {
    let b = B()
    await b.test()
}
```
The following code produces `B` as output, but when the rule is enabled, it suggests to replace to:
```
class B: A {
    func test() { 
        print("B")
    }
}
```
as a result, the output will be `A`, because it doesn't override the original function. So the rule breaks the logic.
The same happens for `throws` in `unneeded_throws_rethrows` rule.

I would suggest excluding the warning for functions that have `override`

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version: 0.63.0
* Xcode version: 26.0.1
* Installation method used: [Universal macOS Binary](https://github.com/realm/SwiftLint/releases/download/0.63.0/portable_swiftlint.zip)
* Configuration file:

```yml
only_rules:
- async_without_await
- unneeded_throws_rethrows
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

**Comment by SimplyDanny** [comment]
For `throws` that's not a problem. Overriding function can omit the `throws` if the function doesn't actually throw. But you're right for `async`. It cannot be left out.
